### PR TITLE
MRPHS-3534 Fetch NIC information of Vms

### PR DESCRIPTION
**Pull request by avnish30jn**
### Symptom:
https://apporbit.atlassian.net/browse/MRPHS-3534

 ### Description:

Fetch NIC information of VMs

### Testing:

**Unit Testing:**
```
[root@deepali-dev3 appos]# go run src/appos/c3ntryrpc/example/c3ntry_client.go
Sending request:
action:"get_template_vm_list" args:"{\"cloud_type\":\"vsphere\", \"Host\":\"209.205.216.11\",\"Username\":\"deepali@vsphere.local\",\"Password\":\"D33p@li@$Dogreat12\",\"Insecure\":true, \"Datacenter\":\"developer-lab\"}"

2017/10/26 08:53:47 Received Response:
result:"[{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":52428800}],\"id\":\"vm-155\",\"name\":\"Template_AO_PROXY\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:66:8b\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":20971520}],\"id\":\"vm-1603\",\"name\":\"centos-3.10.0-327.28.3-dup\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:09:1a\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":15728640}],\"id\":\"vm-1723\",\"name\":\"template-win-12-FRESH\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:71:cc\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-2892\",\"name\":\"IMP-7.2.Provider\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:66:35\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":10485760}],\"id\":\"vm-294\",\"name\":\"ubuntu-16.10-datastore2\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:5a:ec\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":52428800}],\"id\":\"vm-3343\",\"name\":\"IMP-TEMPLATE-entos-7.2\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:4a:35\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":15728640}],\"id\":\"vm-3364\",\"name\":\"template-win-r2-efi\",\"nic_info\":[{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:1c:e5\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-3634\",\"name\":\"rhel-7.2-resource\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:4e:05\",\"nic_name\":\"Network adapter 2\"},{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:01:f7\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-3660\",\"name\":\"TEMPLATE_centos_7.3\",\"nic_info\":[{\"network_name\":\"Test Network\",\"mac_address\":\"00:50:56:89:57:62\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":15728640},{\"name\":\"Hard disk 2\",\"size\":5242880}],\"id\":\"vm-4259\",\"name\":\"template-win-12-r2-2-8\",\"nic_info\":[{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:4a:91\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":41943040}],\"id\":\"vm-4284\",\"name\":\"win-08-r2\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:47:52\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-4745\",\"name\":\"centos7.3_2nics\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:24:1e\",\"nic_name\":\"Network adapter 1\"},{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:6e:59\",\"nic_name\":\"Network adapter 2\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-4769\",\"name\":\"centos_0-nic\",\"nic_info\":null},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":20971520}],\"id\":\"vm-48\",\"name\":\"Template-RHEL-1-disk\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:46:7f\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-4836\",\"name\":\"try_1\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:43:6b\",\"nic_name\":\"Network adapter 1\"},{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:6a:ab\",\"nic_name\":\"Network adapter 2\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-4943\",\"name\":\"RHEl-7.3-noSubscription\",\"nic_info\":[{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:5f:7c\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-4947\",\"name\":\"Rhel-7.3-with-subscr-CLONED\",\"nic_info\":[{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:6a:a0\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-5513\",\"name\":\"centos\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:6f:bf\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":41943040}],\"id\":\"vm-669\",\"name\":\"Template-win-2008-sql_client-old\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:03:7f\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":41943040}],\"id\":\"vm-743\",\"name\":\"sqlclient-domain-2008sql\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:50:e6\",\"nic_name\":\"Network adapter 1\"}]}]" progress:100

2017/10/26 08:53:47 Response String:
{"result":"[{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":52428800}],\"id\":\"vm-155\",\"name\":\"Template_AO_PROXY\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:66:8b\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":20971520}],\"id\":\"vm-1603\",\"name\":\"centos-3.10.0-327.28.3-dup\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:09:1a\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":15728640}],\"id\":\"vm-1723\",\"name\":\"template-win-12-FRESH\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:71:cc\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-2892\",\"name\":\"IMP-7.2.Provider\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:66:35\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":10485760}],\"id\":\"vm-294\",\"name\":\"ubuntu-16.10-datastore2\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:5a:ec\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":52428800}],\"id\":\"vm-3343\",\"name\":\"IMP-TEMPLATE-entos-7.2\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:4a:35\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":15728640}],\"id\":\"vm-3364\",\"name\":\"template-win-r2-efi\",\"nic_info\":[{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:1c:e5\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-3634\",\"name\":\"rhel-7.2-resource\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:4e:05\",\"nic_name\":\"Network adapter 2\"},{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:01:f7\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-3660\",\"name\":\"TEMPLATE_centos_7.3\",\"nic_info\":[{\"network_name\":\"Test Network\",\"mac_address\":\"00:50:56:89:57:62\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":15728640},{\"name\":\"Hard disk 2\",\"size\":5242880}],\"id\":\"vm-4259\",\"name\":\"template-win-12-r2-2-8\",\"nic_info\":[{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:4a:91\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":41943040}],\"id\":\"vm-4284\",\"name\":\"win-08-r2\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:47:52\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-4745\",\"name\":\"centos7.3_2nics\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:24:1e\",\"nic_name\":\"Network adapter 1\"},{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:6e:59\",\"nic_name\":\"Network adapter 2\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-4769\",\"name\":\"centos_0-nic\",\"nic_info\":null},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":20971520}],\"id\":\"vm-48\",\"name\":\"Template-RHEL-1-disk\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:46:7f\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-4836\",\"name\":\"try_1\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:43:6b\",\"nic_name\":\"Network adapter 1\"},{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:6a:ab\",\"nic_name\":\"Network adapter 2\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-4943\",\"name\":\"RHEl-7.3-noSubscription\",\"nic_info\":[{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:5f:7c\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-4947\",\"name\":\"Rhel-7.3-with-subscr-CLONED\",\"nic_info\":[{\"network_name\":\"VM Private\",\"mac_address\":\"00:50:56:89:6a:a0\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":104857600}],\"id\":\"vm-5513\",\"name\":\"centos\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:6f:bf\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":41943040}],\"id\":\"vm-669\",\"name\":\"Template-win-2008-sql_client-old\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:03:7f\",\"nic_name\":\"Network adapter 1\"}]},{\"disks\":[{\"name\":\"Hard disk 1\",\"size\":41943040}],\"id\":\"vm-743\",\"name\":\"sqlclient-domain-2008sql\",\"nic_info\":[{\"network_name\":\"VM Network\",\"mac_address\":\"00:50:56:89:50:e6\",\"nic_name\":\"Network adapter 1\"}]}]","progress":100}

2017/10/26 08:53:47 Breaking..

### CODE REVIEW

File: virtualmachine/vsphere/vm.go [hunk 1]
```diff
@@ -1539,9 +1562,10 @@ func GetTemplateList(vm *VM) ([]map[string]interface{}, error) {
 					}
 				}
 				vmList = append(vmList, map[string]interface{}{
-					"name":  vmo.Name,
-					"id":    vmo.Self.Value,
-					"disks": diskInfo,
+					"name":     vmo.Name,
+					"id":       vmo.Self.Value,
+					"disks":    diskInfo,
+					"nic_info": getNicInfo(vmo),
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/vm.go:1568]: [NTH] Should avoid calling func with struct assignments like this.

File: virtualmachine/vsphere/vm.go [hunk 2]
```diff
@@ -719,6 +725,23 @@ func (vm *VM) RemoveDisk(vmdkFiles []string) error {
 	return nil
 }
 
+// getNicInfo returns the nic info of this VM.
+func getNicInfo(vmMo mo.VirtualMachine) []VirtualEthernetCard {
+	var nicInfo []VirtualEthernetCard
```
Comment (pankajtandon12) [inline virtualmachine/vsphere/vm.go:730]: [NTH] use make() to initialize.

**Review by pankajtandon12 — APPROVED** [review]
LGTM. Minor comments.

**Comment by deepali-gunwant** [comment]
Incorporating NTG comments in new PR....for now UI needs to test this so merging this PR.

**Pull request merged**